src/main/kotlin/io/curity/identityserver/plugin/frejaeid/authentication/representationFunctions.kt

```diff
@@ -0,0 +1,89 @@
+package io.curity.identityserver.plugin.frejaeid.authentication
+
+import se.curity.identityserver.sdk.haapi.*
+import se.curity.identityserver.sdk.haapi.HaapiContract.Links.Relations.QR_CODE
+import se.curity.identityserver.sdk.http.HttpMethod
+import se.curity.identityserver.sdk.http.MediaType
+import se.curity.identityserver.sdk.web.LinkRelation
+import se.curity.identityserver.sdk.web.Representation
+import java.net.URI
+
+class GetRepresentationFunction : RepresentationFunction
+{
+    companion object
+    {
+        val titleMessage: Message = Message.ofKey("view.title")
+        val submitMessage: Message = Message.ofKey("view.submit")
+    }
+    override fun apply(model: RepresentationModel, factory: RepresentationFactory): Representation
+    {
+        val authUrl = URI.create(model.getString("_authUrl"))
+        val userInfoType = model.getString("userInfoType")
+
+        return factory.newAuthenticationStep { step ->
+            step.addMessage(Message.ofKey("view.$userInfoType.description"), HaapiContract.MessageClasses.INFO)
+            step.addFormAction(HaapiContract.Actions.Kinds.LOGIN, authUrl, HttpMethod.POST,
+                    MediaType.X_WWW_FORM_URLENCODED, titleMessage, submitMessage) { fields ->
+                fields.addTextField("username", Message.ofKey("view.$userInfoType.label"))
+
+            }
+        }
+    }
+}
+
+class ErrorRepresentationFunction : RepresentationFunction
+{
+    companion object
+    {
+        val restartMessage: Message = Message.ofKey("error.restart")
+    }
+    override fun apply(model: RepresentationModel, factory: RepresentationFactory): Representation
```

Comment:
Note that the `apply` is only called if the status is `2xx`. For `4xx`/`5xx`, we need to implement `applyError` if we need a custom behavior (there is a default behavior that is applies automatically and that produces a `problem+json` representation).

Reply:
This representation function seems to also be used with success responses (`WaitRequestHandler.kt:245`)